dotnet-desktop-guide/net/wpf/documents/snippets/printing-overview/csharp/MainWindow.xaml.cs

```diff
@@ -0,0 +1,166 @@
+                DefaultExt = ".xps",
+                Filter = "Text documents (.xps)|*.xps"
+            };
+
+            // Show open file dialog box.
+            bool? result = openFileDialog.ShowDialog();
+
+            // Process open file dialog box results.
+            if (result == true)
+            {
+                // Show selected XPS file path.
+                txtXpsFilePath.Text = openFileDialog.FileName;
+            }
+        }
+
+        private void BtnPrint_Click(object sender, RoutedEventArgs e)
+        {
+            // Check the XPS file exists.
+            if (!File.Exists(txtXpsFilePath.Text))
+            {
+                MessageBox.Show("First select an XPS file.");
+                return;
+            }
+
+            // Get the print queue for the selected printer.
+            PrintQueue selectedPrintQueue = GetPrintQueues().SingleOrDefault(x => Equals(x.Name, cmbPrinters.SelectedItem.ToString()));
+
+            // Get a print ticket with modified print options for the selected printer.
+            PrintTicket printTicket = GetPrintTicket(selectedPrintQueue);
+
+            // Send the XPS document to the selected printer.
+            PrintXpsDocumentAsync(txtXpsFilePath.Text, selectedPrintQueue, printTicket);
+
+            // Show modified print options.
+            lblPrintTicketOptions.Content = "";
+            List<string> optionList = new();
+            if (printTicket.Collation == Collation.Collated)
+                optionList.Add("Collated");
+            if (printTicket.Duplexing == Duplexing.TwoSidedLongEdge)
+                optionList.Add("TwoSidedLongEdge");
+            if (printTicket.Stapling == Stapling.StapleDualLeft)
+                optionList.Add("StapleDualLeft");
+            lblPrintTicketOptions.Content = string.Join(", ", optionList);
+
+            // Show result message.
+            MessageBox.Show($"Sent {Path.GetFileName(txtXpsFilePath.Text)} to the printer.");
+        }
+
+        // <SampleCode2>
+        /// <summary>
+        /// Return a collection of print queues, which individually hold the features or states
+        /// of a printer as well as common properties for all print queues.
+        /// </summary>
+        /// <returns>A collection of print queues.</returns>
+        public static PrintQueueCollection GetPrintQueues()
+        {
+            // Create a LocalPrintServer instance, which represents 
+            // the print server for the local computer.
+            LocalPrintServer localPrintServer = new();
+
+            // Get all print queues on the local computer.
+            PrintQueueCollection printQueueCollection = localPrintServer.GetPrintQueues();
+
+            // Return a collection of print queues, which individually hold the features or states
+            // of a printer as well as common properties for all print queues.
+            return printQueueCollection;
+        }
+        // </SampleCode2>
+
+        // <SampleCode1>
```

Comment:
Just FYI. In the future you can be more specific with the snippet ID. I generally use the method name:
`<GetPrintTicket>` which makes it clearer when browsing the source of the article.

Reply:
Noted, thanks.